### PR TITLE
Node render improvements

### PR DESCRIPTION
**Pull request by evgenoid**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
No way to detect selected node or not on render
No way to hide node selection circle

Issue Number: N/A

## What is the new behavior?
Add selected property to renderNode callback
Add ability to hide node selection.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

Screenshot 2024-04-19 at 13 13 37

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "reagraph",
-  "version": "4.16.1",
+  "version": "4.16.2",
```
Comment (amcdnl) [inline package.json:3]: versoins should be seperate from PRs for revert ease

File: src/types.ts [hunk 2]
```diff
@@ -232,6 +232,11 @@ export interface NodeRendererProps {
    */
   active: boolean;
 
+  /**
+   * Whether the node is selected or not.
+   */
+  selected?: boolean;
```
Comment (amcdnl) [inline src/types.ts:238]: I don't see this used anywhere

**Review by amcdnl — CHANGES_REQUESTED** [review]
I'm not sure this is the right way to do this. I need to think on it but initial thoughts might be moving the ring into the node so a user can omit it if they want rather than have a prop like this.

File: src/symbols/Node.tsx [hunk 3]
```diff
@@ -261,7 +267,8 @@ export const Node: FC<NodeProps> = ({
           active: combinedActiveState,
           opacity: selectionOpacity,
           animated,
-          node
+          node,
+          selected: isSelected
```
Comment (evgenoid) [inline src/symbols/Node.tsx:271]: >I don't see this used anywhere

Here

**Comment by evgenoid** [comment]
> I'm not sure this is the right way to do this. I need to think on it but initial thoughts might be moving the ring into the node so a user can omit it if they want rather than have a prop like this.

Ok, let me know what you decided.

File: src/symbols/Node.tsx (hunk 3 above)
Comment (amcdnl) [reply src/symbols/Node.tsx:271]: You need to pull the selected from the state slice for perf reasons

Comment (evgenoid) [reply src/symbols/Node.tsx:271]: But isSelected uses everywhere in this component.

Comment (amcdnl) [reply src/symbols/Node.tsx:271]: Nevermind I was confused by the `selected` type addition

**Comment by evgenoid** [comment]
@amcdnl any updates?

**Pull request closed without merging**